app/controllers/v0/sessions_controller.rb

```diff
@@ -133,23 +134,27 @@ def user_logout(saml_response)
       end
     end
 
-    # rubocop:disable Metrics/MethodLength
+    def success_stat_tags(saml_response, *tags)
```

Comment:
This method does a lot of string-building and is accessing a lot of methods on `@current_user` and `saml_response` and since it's only used in one place I thought this might be better as a variable.  But when I plug it into `login_stats`, the rubocop warning comes back for MethodLength.  Which lead me to thinking maybe just all the successful login information could be it's own method:
```
    def login_stats_success(saml_response, tracker)
      success_tags = [
        "loa:#{@current_user.loa[:current]}",
        "idp:#{@current_user.identity.sign_in[:service_name]}",
        "context:#{saml_response.authn_context}",
        VERSION_TAG
      ]
      StatsD.increment(STATSD_LOGIN_NEW_USER_KEY, tags: [VERSION_TAG]) if request_type == 'signup'
      # track users who have a shared sso cookie
      StatsD.increment(STATSD_LOGIN_SHARED_COOKIE, tags: success_tags) if cookies.key?(Settings.sso.cookie_name)
      StatsD.increment(STATSD_LOGIN_STATUS, tags: success_tags + ['status:success'])
      StatsD.measure(STATSD_LOGIN_LATENCY, tracker.age, tags: success_tags)
    end
```

Then below becomes:
```
def login_stats(status, saml_response, user_session_form = nil)
      tracker = url_service(user_session_form&.saml_uuid).tracker
      case status
      when :success
        login_stats_success(saml_response, tracker)
        callback_stats(:success, saml_response)
      when :failure
[...]
```
What do you think about this approach?

Reply:
Similarly in `app/controllers/v1/sessions_controller.rb` if you like it

Reply:
yeah that works.  The only reason I broke it out in the first place was to get around the robocop warnings